packages/block-editor/src/components/image-editor/use-save-image.js

```diff
@@ -1,6 +1,8 @@
 /**
  * WordPress dependencies
  */
+// Disable Reason: Needs to be refactored.
+// eslint-disable-next-line no-restricted-imports
 import apiFetch from '@wordpress/api-fetch';
```

Comment:
@talldan No urgency here but it would be good to find a solution for this. (move it to editor or something)

Reply:
@youknowriad I don't think this was anything to do with me, btw, but happy to have a think about it and help out where I can. 

@ajlende originally implemented the feature, and [then moved it to the block editor package](https://github.com/WordPress/gutenberg/pull/31607), my role was only [some refactoring back when it was still in the block-library package](https://github.com/WordPress/gutenberg/pull/27089).

I don't think it can be moved to `editor` as there's multiple blocks using the component, and they can't import it from editor.

One thought is that maybe the API Fetch part can be moved out to the specific blocks, but might need some back compat, as plugins are using this (WooCommerce being the main one I can see). Not sure how easy that would be to do.

Or it could potentially be implemented in a similar way to `LinkControl` / `__experimentalFetchLinkSuggestions` where the `fetch` code is an editor setting, and then the component uses that. It doesn't necessarily guarantee back compat though, it won't cover if that component is used outside of a block editor.

Will keep thinking on it.

Reply:
Sorry for the wrong ping and yeah I agree that these are not always easy problems to tackle but I do believe important enough for the long term. 

Shouldn't this component be using the mediaUpload setting passed to the BlockEditorProvider instead?

Reply:
Perhaps, at a first look it seems like `mediaUpload` / `uploadMedia` uses the `/wp/v2/media` endpoint and can upload multiple files, while this one uses `/wp/v2/media/${ id }/edit` endpoint and is just for modifying one file.

But I think that's a good call, maybe a new setting/function that works in a similar way to `mediaUpload` could be an option, or adapting `mediaUpload` so that it can handle edits.

Reply:
or maybe we can evolve the same function to work differently depending on whether you pass an id in the object or not.